packages/cli/src/lib/new/prompts.ts

```diff
@@ -0,0 +1,152 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import inquirer from 'inquirer';
+import { Prompt, ConfigurablePrompt } from './types';
+import { parseOwnerIds } from '../codeowners';
+
+export function pluginIdPrompt(): Prompt<{ id: string }> {
+  return {
+    type: 'input',
+    name: 'id',
+    message: 'Enter the ID of the plugin [required]',
+    validate: (value: string) => {
+      if (!value) {
+        return 'Please enter the ID of the plugin';
+      } else if (!/^[a-z0-9]+(-[a-z0-9]+)*$/.test(value)) {
+        return 'Plugin IDs must be lowercase and contain only letters, digits, and dashes.';
+      }
+      return true;
+    },
+  };
+}
+
+export function moduleIdIdPrompt(): Prompt<{ moduleId: string }> {
+  return {
+    type: 'input',
+    name: 'moduleId',
+    message: 'Enter the ID of the module [required]',
+    validate: (value: string) => {
+      if (!value) {
+        return 'Please enter the ID of the module';
+      } else if (!/^[a-z0-9]+(-[a-z0-9]+)*$/.test(value)) {
+        return 'Module IDs must be lowercase and contain only letters, digits, and dashes.';
+      }
+      return true;
+    },
+  };
+}
+
+export function npmRegistryPrompt(): Prompt<{ npmRegistry: string }> {
+  return {
+    type: 'input',
+    name: 'npmRegistry',
+    message: 'Please specify your NPM registry [optional]',
+    validate: (value: string) => {
+      if (!value) {
+        return 'Please enter the URL of your NPM registry';
+      } else if (!/^http*$/.test(value)) {
+        return 'Invalid URL.';
+      }
+      return true;
+    },
+  };
+}
+
+export function ownerPrompt(
+  codeOwnersPath: string | undefined,
+): Prompt<{ owner?: string }> {
+  return {
+    type: 'input',
+    name: 'owner',
+    message: 'Enter an owner to add to CODEOWNERS [optional]',
+    when: Boolean(codeOwnersPath),
+    validate: (value: string) => {
+      if (!value) {
+        return true;
+      }
+
+      const ownerIds = parseOwnerIds(value);
+      if (!ownerIds) {
+        return 'The owner must be a space separated list of team names (e.g. @org/team-name), usernames (e.g. @username), or the email addresses (e.g. user@example.com).';
+      }
+
+      return true;
+    },
+  };
+}
+
+export async function promptOptions({
+  prompts,
+  globals,
+  codeOwnersFilePath,
+}: {
+  prompts: ConfigurablePrompt[];
+  globals: Record<string, string>;
+  codeOwnersFilePath: string | undefined;
+}): Promise<Record<string, string>> {
+  const answers = await inquirer.prompt(
+    prompts.map((prompt: ConfigurablePrompt) => {
+      if (typeof prompt === 'string') {
+        switch (prompt) {
+          case 'id':
+            return pluginIdPrompt();
+          case 'moduleId':
+            return moduleIdIdPrompt();
+          case 'npmRegistry':
+            return npmRegistryPrompt();
```

Comment:
This should probably just be a configuration option right? defaulted to whatever is in `.yarn.yml` / `.npmrc` too